language-design.md

```diff
@@ -18,6 +18,53 @@ For each quote, describe what it was about the quote that led you pick it.
+of whom are renowned computer scientists for whom I have tremendous respect!
+But a computer scientist is not the same as a software engineer." [Smith, 2016]
+
+We picked this quote because while most of what we have been reading has been 
+from professors and researchers who study the design and implementation of 
+languages, we rarely get a voice from an "outsider" in this respect - someone 
+who just wants to use a language for a purpose and not to analyze its history 
+and design decisions. We think the goal of a well-designed and "good" DSL is to 
+be a tool that accomplishes the domain-specific goals easily or intuitively. 
+Smith does not think R is not well-designed, and traces the cause to the lack 
+of software engineers contributing to R.
+
+The author specifically makes the distinction between computer scientist and 
+software engineers, which brings up differences between academia and industry. 
+However, many languages developed for and used by academia are very planned 
+and are implemented very specifically, i.e. Prolog and Lisp. Languages 
+that are considered better designed than R also have a lot of contributions 
+from computer scientists who are not software engineers, so it doesn't seem 
+like that is the reason why R is badly designed.
+
+Doing some research, it seems that R was derived from S, with some language 
+components inspired by Scheme. It was developed by two people initially, and 
+now its design is directed by the R-core of around twenty people. It is unclear 
+how many of those people still actively contribute, and maybe that is the 
+problem. Maybe because R is open-source, a kind of a project built as a bazaar, 
+it incorporated too many different designs and features that, when looked at 
+together by a newcomer, seems like a series of hacks and a mishmash of things. 
+
+2. “The key point is that in the bazaar style of building a program or designing a language or what you will, the plan can change in real time to meet the needs of those who work on it and use it. This tends to make users stay with it as time goes by; they will take joy in working hard and helping out if they know that their wants and needs have some weight and their hard work can change the plan for the better.”
+
+
+We chose this quote because we feel that Steele’s point seems to be a very true pattern across many of the popular languages today. A typical example of this would be Javascript, which was created around 20 years ago within the span of 10 days. It was meant as a sort of “glue” language to build web apps, with the “components” being written in Java. Fast forward a decade and a half and so many useful open-source libraries have been created, some of which have widespread communities. Who could have predicted early on that we can eventually use JavaScript on the server-side? Meanwhile, languages that did not allow for this style of open-source, pivoting growth like Pascal and PL/I are now the stuff of history books. 
```

Comment:
I also agree that Javascript is based on an open-source "bazaar" design. But that doesn't necessarily mean bazaars grow more than cathedrals. Java is a good counterexample.

One factor that may expedite growth is the popularity of certain tech industries. With the rise of web and mobile app development, it isn't surprising that Java and Javascript grew as its user base added features and provided feedback. If for some odd reason web development was no longer popular, I think Javascript would stop growing as well.